connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -186,6 +192,8 @@ public boolean alterOffsets(Map<String, String> connectorConfig, Map<Map<String,
 
     private void refreshConsumerGroups()
             throws InterruptedException, ExecutionException {
+        // Checking null to avoid NPE.
+        Set<String> knownConsumerGroups = Objects.isNull(this.knownConsumerGroups) ? Collections.emptySet() : this.knownConsumerGroups;
```

Comment:
1. Same nit RE `Objects::isNull`; we can probably do something like this instead: `this.knownConsumerGroups != null ? this.knownConsumerGroups : Collections.emptySet()`
2. I agree that it's good to safeguard against NPEs, but maybe we can elaborate in the comment why this is necessary: if `loadInitialConsumerGroups` fails for some reason, then `knownConsumerGroups` may be null, and we still want this method to be able to recover gracefully.

Reply:
That makes sense. The current comment is too brief to convey why the NPE check is needed. To understand it, one would have to read the code and grasp how it works. I will make the comment more specific to clarify its necessity.

Thank you for pointing that out. 🙇🏼